ui/src/components/filters/QasFilters.vue

```diff
@@ -155,15 +166,25 @@ export default {
     },
 
     formattedFieldsProps () {
-      const fieldsProps = {}
+      const formattedFieldsProps = {}
 
-      for (const key in this.fieldsProps) {
+      for (const key in this.fields) {
         const decamelizedFieldKey = decamelize(key)
-
-        fieldsProps[decamelizedFieldKey] = { ...this.fieldsProps[key] }
+        const fieldsProps = this.fieldsProps[key] || {}
+        const useLazyLoading = this.fields[key].useLazyLoading || fieldsProps.useLazyLoading
+
+        formattedFieldsProps[decamelizedFieldKey] = {
+          ...fieldsProps,
+          ...(useLazyLoading && {
```

Comment:
Aqui é onde se encontra a tratativa para campos lazy loading. Eu faço um `for` por todos os `fields` do filtro, e realizo a formatação dos fieldsProps. Passando os fieldsProps que foram passados externamente como prop, e atribuindo também mais algumas propriedades se o campo for lazy loading.

No caso se o campo for lazy loading, atribuo a prop ` useFetchOptionsOnFocus` (nova prop) por padrão, isso porque o menu do filtro toda vez que é aberto é criado uma nova instância e quando é fechado é matado aquela instância (é semelhante ao funcionamento de um dialog no Quasar), o que causa que toda vez que abre novamente ele bateria todos os endpoints de lazy loading novamente. Acredito que isso seria bem ruim pra performance, por isso, aqui vem uma mudança de comportamento, que seria bater o endpoint de lazy loading somente ao dar focus no campo. Acredito que essa mudança de comportamento poderia ser válida para todos os outros casos de lazy loading no sistema, depois queria ter essa conversa com vocês.

Por fim, escuto pelo evento `onUpdate:selectedOptions`, esse evento irá me trazer as opções selecionadas e com essas opções vou salvar naquele objeto auxiliar que expliquei anteriormente.